CourseChamp/src/test/java/ca/mcgill/ecse428/CourseChamp/Service/ReviewServiceTests.java

```diff
@@ -0,0 +1,64 @@
+package ca.mcgill.ecse428.CourseChamp.Service;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import ca.mcgill.ecse428.CourseChamp.exception.CourseChampException;
+import ca.mcgill.ecse428.CourseChamp.model.Review;
+import ca.mcgill.ecse428.CourseChamp.repository.ReviewRepository;
+import ca.mcgill.ecse428.CourseChamp.service.ReviewService;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.HttpStatus;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReviewServiceTests {
+
+    @Mock
+    private ReviewRepository reviewRepository;
+
+    @InjectMocks
+    private ReviewService reviewService;
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    public void testFindReviewsByCourseCodeSuccess() {
+        String courseCode = "ECSE428";
+        List<Review> mockReviews = new ArrayList<>();
+        mockReviews.add(new Review()); // Add a mock Review instance
+
+        when(reviewRepository.findReviewsByCourseCode(courseCode)).thenReturn(mockReviews);
+
+        List<Review> reviews = reviewService.findReviewsByCourseCode(courseCode);
+
+        assertNotNull(reviews, "The returned review list should not be null.");
+        assertFalse(reviews.isEmpty(), "The review list should not be empty.");
+        assertEquals(mockReviews.size(), reviews.size(), "The size of returned review list should match the mock list.");
+        verify(reviewRepository).findReviewsByCourseCode(courseCode);
+    }
```

Comment:
I don't understand how this works, your mockReview are never pushed to the repo... right??

Reply:
They're not pushed to the repo we're mocking the repo with the when() annotation

Reply:
Yes but isn't your mock repo empty?

Reply:
Yeah but the goal of the tests is to simulate the behavior of the methods in the repo. in the when for the success find review I create an object of type review with a course code "course code" and check that it returns an object review with a course code "course code". We're testing the behavior of the code